### PR TITLE
Dockerize gumbler!

### PR DESCRIPTION
**Pull request by claudijd**
This is not done nor is it tested in any way, so it will need a few tweaks, but if you want to chase #9 and #10 this is a good start from an infra perspective

### CODE REVIEW

File: docker/web [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+RUN apt-get update && \
+    apt-get install -y git python-pip curl && \
+    cd /app && \
+    git clone https://github.com/BuffaloWill/gumbler.git && \
+    cd gumbler && \
+    pip install -r requirements.txt && \
+    pip install requests && \
+    mkdir projects && \
+    cd projects && \
+    git clone git clone https://github.com/BuffaloWill/gumbler.git
```
Comment (claudijd) [inline docker/web]: Yes, this is inception, but maybe you will just give a user a config file or something to list all the projects they want to gumblerize that would make this cleaner.

**Comment by claudijd** [comment]
To run it, just run "docker-compose up", but I'm still working out the buggies.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+services:
+  database:
+    image: mongo:latest
+    environment:
+      - MONGO_DATA_DIR=/data/db
+      - MONGO_LOG_DIR=/dev/null
+    networks:
+      - default
+    volumes:
+      - ./data/db:/data/db
```
Comment (claudijd) [inline docker-compose.yml:11]: You can nuke the volumes reference if you don't want persistent DB

**Comment by claudijd** [comment]
This as of now will get you here.....

Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)

With the inception project choice, what's left is getting flask to listen on 0.0.0.0 in th docker container

**Comment by claudijd** [comment]
```
$ docker-compose up
Starting gumbler_database_1 ... 
Starting gumbler_database_1 ... done
Starting gumbler_web_1 ... 
Starting gumbler_web_1 ... done
Attaching to gumbler_database_1, gumbler_web_1
database_1  | 2017-07-18T19:59:25.664+0000 I CONTROL  [initandlisten] MongoDB starting : pid=1 port=27017 dbpath=/data/db 64-bit host=dc9fd243f84a
database_1  | 2017-07-18T19:59:25.664+0000 I CONTROL  [initandlisten] db version v3.4.4
database_1  | 2017-07-18T19:59:25.664+0000 I CONTROL  [initandlisten] git version: 888390515874a9debd1b6c5d36559ca86b44babd
database_1  | 2017-07-18T19:59:25.665+0000 I CONTROL  [initandlisten] OpenSSL version: OpenSSL 1.0.1t  3 May 2016
database_1  | 2017-07-18T19:59:25.665+0000 I CONTROL  [initandlisten] allocator: tcmalloc
database_1  | 2017-07-18T19:59:25.665+0000 I CONTROL  [initandlisten] modules: none
database_1  | 2017-07-18T19:59:25.665+0000 I CONTROL  [initandlisten] build environment:
database_1  | 2017-07-18T19:59:25.665+0000 I CONTROL  [initandlisten]     distmod: debian81
database_1  | 2017-07-18T19:59:25.665+0000 I CONTROL  [initandlisten]     distarch: x86_64
database_1  | 2017-07-18T19:59:25.665+0000 I CONTROL  [initandlisten]     target_arch: x86_64
database_1  | 2017-07-18T19:59:25.665+0000 I CONTROL  [initandlisten] options: {}
database_1  | 2017-07-18T19:59:25.671+0000 I -        [initandlisten] Detected data files in /data/db created by the 'wiredTiger' storage engine, so setting the active storage engine to 'wiredTiger'.
database_1  | 2017-07-18T19:59:25.673+0000 I STORAGE  [initandlisten] wiredtiger_open config: create,cache_size=487M,session_max=20000,eviction=(threads_min=4,threads_max=4),config_base=false,statistics=(fast),log=(enabled=true,archive=true,path=journal,compressor=snappy),file_manager=(close_idle_time=100000),checkpoint=(wait=60,log_size=2GB),statistics_log=(wait=0),
database_1  | 2017-07-18T19:59:27.129+0000 I CONTROL  [initandlisten] 
database_1  | 2017-07-18T19:59:27.129+0000 I CONTROL  [initandlisten] ** WARNING: Access control is not enabled for the database.
database_1  | 2017-07-18T19:59:27.129+0000 I CONTROL  [initandlisten] **          Read and write access to data and configuration is unrestricted.
database_1  | 2017-07-18T19:59:27.129+0000 I CONTROL  [initandlisten] 
database_1  | 2017-07-18T19:59:27.148+0000 I FTDC     [initandlisten] Initializing full-time diagnostic data capture with directory '/data/db/diagnostic.data'
database_1  | 2017-07-18T19:59:27.148+0000 I NETWORK  [thread1] waiting for connections on port 27017
web_1       |  * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
```

**Comment by claudijd** [comment]
OK, so here's the deets on what you need to do once you change code, forgot this for a bit, so it required some head and desk banging....

`docker-compose build` <= rebuilds the image and re-imports code changes
`docker-compose up` <= starts up the image you built

**Comment by BuffaloWill** [comment]
Thanks @claudijd!

**Pull request merged**